core/procs.go

```diff
@@ -1331,18 +1331,20 @@ var procArrayMap Proc = func(args []Object) Object {
 	return res
 }
 
+const bufferHashMask uint32 = 0x5ed19e84
+
 var procBuffer Proc = func(args []Object) Object {
 	if len(args) > 0 {
 		s := EnsureString(args, 0)
-		return &Buffer{bytes.NewBufferString(s.S)}
+		return MakeBuffer(bytes.NewBufferString(s.S), s.Hash()^bufferHashMask)
```

Comment:
Since `Buffer` object are mutable, I don't think it's right to assign them a hash value based on their initial content...

Reply:
Indeed, I was being overzealous trying to find something "stable". Besides the fact that there's no reason to assume this would be stable (the initial content could easily differ from run to run), it'd result in different buffers having identical hashes merely because they started out with the same content (which could be the empty string, versus a nil argument).